File: src/main/java/gargoyle/sexbomb/util/applet/AppletContextImpl.java
package gargoyle.sexbomb.util.applet;

import java.awt.Desktop;
import java.io.IOException;
import java.net.URISyntaxException;
import java.net.URL;

class AppletContextImpl implements AppletContext {
    @Override
    public void showDocument(URL url) {
        try {
            Desktop.getDesktop().browse(url.toURI());
        } catch (IOException | URISyntaxException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void showDocument(URL url, String target) {
        try {
            Desktop.getDesktop().browse(url.toURI());
        } catch (IOException | URISyntaxException e) {
            e.printStackTrace();
        }
    }
}

File: src/main/kotlin/gargoyle/sexbomb/util/applet/AppletContextImpl.kt
package gargoyle.sexbomb.util.applet

import java.awt.Desktop
import java.io.IOException
import java.net.URISyntaxException
import java.net.URL

internal class AppletContextImpl : AppletContext {
    override fun showDocument(url: URL) {
        try {
            Desktop.getDesktop().browse(url.toURI())
        } catch (e: IOException) {
            e.printStackTrace()
        } catch (e: URISyntaxException) {
            e.printStackTrace()
        }
    }

    override fun showDocument(url: URL, target: String) {
        try {
            Desktop.getDesktop().browse(url.toURI())
        } catch (e: IOException) {
            e.printStackTrace()
        } catch (e: URISyntaxException) {
            e.printStackTrace()
        }
    }
}
